### PR TITLE
fix(controller): handle CNI config mismatch on underlay recreate

### PR DESCRIPTION
**Pull request by fedepaol**
**Is this a BUG FIX or a FEATURE ?**:

> Uncomment only one, leave it on its own line:
>
/kind bug
> /kind cleanup
> /kind feature
> /kind design
> /kind flake
> /kind failing
> /kind documentation
> /kind regression
> /kind example

**What this PR does / why we need it**:

Error found in   [FAIL] Routes between bgp and the fabric with Underlay in ipv4 MacvlanDHCP with vnis and frr-k8s [BeforeEach] translates EVPN incoming routes as BGP routes

https://github.com/openperouter/openperouter/actions/runs/30977939400/job/92217350377?pr=509

When an underlay is deleted and recreated with a different CNI config (e.g. static -> dhcp IPAM) for the same interface name, the reconciler can miss the cleanup path. The test deletes old underlays and creates new ones sequentially, but the reconciler is async: by the time it processes the queued delete event, getConfigFromAPI() reads current state which already has the new underlays. Since len(config.Underlays) > 0, restoreUnderlay() is never called and the stale CNI cache entry is never removed.

From the CI logs on worker2: the reconcile for underlay-cni-1 (deleted) found underlay-dhcp-* already in the API. SetupUnderlay tried to provision "underlay0" with DHCP config but the cache held the old static config, producing 265 ConfigMismatchError loops that prevented VRF creation and blocked all BGP sessions on that node.

Handle this by deleting the old attachment and retrying Add when a config mismatch is detected.

**Special notes for your reviewer**:

The way we handle the scenario is debatable, and the scenario itself is related to a very specific edge case which is, quick deletion and recreation of two underlays with the same interface name. We could probably workaround this by using a different interface name per test

**Release note**:

```release-note
Bug: if the delete of a previous underlay and the add of a new one are squashed together, and both use CNI but with a different configuration, the underlay is not processed.
```

**AI Guidelines Acknowledgment**:

- [x] I have reviewed all changes in this PR, including any AI-generated content, and I take full responsibility for its accuracy and correctness.

## Summary by CodeRabbit

* **Bug Fixes**
  * Automatically detects mismatched network interface configurations.
  * Removes stale interface settings and provisions the interface again with the updated configuration.
  * Reports errors if cleanup or re-provisioning fails.

* **Tests**
  * Added coverage for updating an interface after its configuration changes.

### CODE REVIEW

File: internal/hostnetwork/underlay_cnidev_test.go [hunk 1]
```diff
@@ -243,6 +243,33 @@ var _ = Describe("Underlay CNI configuration", func() {
 		Expect(loggedCommands()).To(ConsistOf("ADD net1", "DEL net1"))
 	})
 
+	It("rebuilds the cni interface when the config changes", func() {
```
Comment (maiqueb) [inline internal/hostnetwork/underlay_cnidev_test.go:246]: I think we literally said we don't want to handle configuration changes in CNI interfaces in the enhancement. 

While I understand that in an eventually consistent world a DELETE + a CREATE (the same key, but with different config) is an UPDATE, and that our docs instruct the user to re-create the configuration, I think we need to understand better what we want to do.
